src/AnalyzerPowerPack/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -186,7 +234,34 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
 
             private bool IsSerializable(ITypeSymbol namedTypeSymbol)
             {
-                return namedTypeSymbol.GetAttributes().Any(a => a.AttributeClass == _serializableAttributeTypeSymbol);
+                return IsPrimitiveType(namedTypeSymbol) ||
+                       namedTypeSymbol.SpecialType == SpecialType.System_String ||
+                       namedTypeSymbol.GetAttributes()
+                           .Any(a => a.AttributeClass.Equals(_serializableAttributeTypeSymbol));
+            }
+
+            private static bool IsPrimitiveType(ITypeSymbol type)
+            {
+                switch (type.SpecialType)
+                {
+                    case SpecialType.System_Boolean:
+                    case SpecialType.System_Byte:
```

Comment:
Decimal is not a primitive type (the CLR doesn't understand decimals - it's a c# keyword). So this method shouldn't include it but we probably don't want to complain about decimal fields and so the check should be next to the string check.